### PR TITLE
[13.0][FIX] account_global_discount: Add option to discount apply when line contains only one tax with zero amount

### PR DESCRIPTION
**Pull request by victoralmau**
Add option to discount apply when line contains only one tax with zero amount.

@Tecnativa TT27911

### CODE REVIEW

**Comment by victoralmau** [comment]
Error happen in `account_invoice_pricelist` tests, @manuelregidor can you review it please?

File: account_global_discount/models/account_move.py [hunk 1]
```diff
@@ -131,6 +132,33 @@ def _set_global_discounts_by_tax(self):
                     )
                 )
                 base = discount["base_discounted"]
+        # Check all moves with tax_ids defined to check if any not discount created
+        # (tax amount is zero and only one tax is aplly)
+        if self.line_ids.filtered("tax_ids"):
```
Comment (ernestotejeda) [inline account_global_discount/models/account_move.py]: I think we can remove this 'if' statement

File: account_global_discount/models/account_move.py [hunk 2]
```diff
@@ -131,6 +132,33 @@ def _set_global_discounts_by_tax(self):
...
+            for move in self.line_ids.filtered("tax_ids"):
+                key = tuple(move.mapped("tax_ids").ids)
+                if key in taxes_keys:
+                    if taxes_keys[key]:
```
Comment (ernestotejeda) [inline account_global_discount/models/account_move.py]: These two lines can be written in one 'if' statement: `if taxes_keys.get("key"):`

**Review by ernestotejeda — APPROVED** [review]
Just two non-blocking details.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
We need to fix this problem, so I merge manually, and let the unrelated error in `account_invoice_pricelist`.

**Pull request merged**